CANTroller2/src/utils.h

```diff
@@ -11,7 +11,35 @@
 #define adcrange_adc 4095  // = 2^adcbits-1
 #define adcmidscale_adc 2047  // = 2^(adcbits-1)-1
 
-// pin operations that first check if pin exists for the current board
+/* Utility Functions */
+#define arraysize(x) ((int32_t)(sizeof(x) / sizeof((x)[0])))  // A macro function to determine the length of string arrays
+#define floor(amt, lim) ((amt <= lim) ? lim : amt)
+#define ceiling(amt, lim) ((amt >= lim) ? lim : amt)
+
+#undef max // NOTE: whats wrong with the builtin...?

```

Comment:
agreed, we can 
#include <Arduino.h> 
and use min(), max(), map(), floor(), ceiling(), arraysize(), and constrain()
We do still need our ema_filt though

Reply:
I wonder if there's a reason soren wasn't using them...?

Reply:
changed to a TODO